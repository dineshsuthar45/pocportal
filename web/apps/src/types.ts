import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
