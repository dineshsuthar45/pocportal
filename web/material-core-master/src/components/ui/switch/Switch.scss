.switch-root {
	.MuiSwitch-root {
		.MuiSwitch-track {
			opacity: 1;
		}

		.MuiSwitch-switchBase:hover:not(.Mui-checked) + .MuiSwitch-track {
			background-color: var(--m-colour-greyScale-50);
		}
		.MuiSwitch-switchBase {
			box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 3px rgba(0, 0, 0, 0.12);
		}
	}

	.MuiSwitch-track {
		background-color: var(--m-colour-greyScale-200);
	}

	&.dark {
		.MuiSwitch-root {
			.MuiSwitch-switchBase:hover:not(.Mui-checked) + .MuiSwitch-track {
				background-color: var(--m-colour-greyScale-500);
			}
			.MuiSwitch-switchBase {
				box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14),
					0px 1px 3px rgba(0, 0, 0, 0.12);
			}
		}

		.MuiSwitch-track {
			background-color: var(--m-colour-greyScale-600);
		}
	}
}
