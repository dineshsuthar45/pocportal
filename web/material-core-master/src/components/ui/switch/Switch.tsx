import { useTheme } from '@mui/material';
import MuiSwitch from '@mui/material/Switch';
import classNames from 'classnames';
import { ChangeEvent, useEffect, useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { MFormControlLabel } from '../internal/formControlLabel';
import './Switch.scss';
import { SwitchProps } from './Switch.types';

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/switch/
 */
export const MSwitch = ({
	label,
	labelPlacement = 'end',
	tooltipTitle = '',
	checked = false,
	color = 'secondary',
	disabled = false,
	disableRipple = false,
	edge = false,
	id = uuidv4(),
	required = false,
	size = 'medium',
	...props
}: SwitchProps) => {
	const theme = useTheme();
	const [internalTooltipTitle, setInternalTooltipTitle] = useState(tooltipTitle || label || '');
	const [showAlwaysTooltip, setShowAlwaysTooltip] = useState(false);

	useEffect(() => {
		setInternalTooltipTitle(tooltipTitle || label || '');
	}, [tooltipTitle, label]);

	useEffect(() => {
		setShowAlwaysTooltip(!!tooltipTitle && tooltipTitle !== label);
	}, [label, tooltipTitle]);

	const handleClick = (event: unknown) => {
		if (!disabled && props.onChange) {
			props.onChange(event as ChangeEvent<HTMLInputElement>, !checked);
		}
	};

	return (
		<div onClick={handleClick} className={classNames('switch-root', theme.palette.mode)}>
			<MFormControlLabel
				label={label}
				labelPlacement={labelPlacement}
				tooltipProps={{
					title: internalTooltipTitle,
					showAlways: showAlwaysTooltip,
				}}
				disabled={disabled}
			>
				<MuiSwitch
					id={id}
					checked={checked}
					color={color}
					disabled={disabled}
					size={size}
					{...props}
					onChange={() => {}}
				/>
			</MFormControlLabel>
		</div>
	);
};

MSwitch.defaultProps = {
	labelPlacement: 'end',
	tooltipTitle: '',
	checked: false,
	color: 'secondary',
	disabled: false,
	disableRipple: false,
	edge: false,
	id: uuidv4(),
	required: false,
	size: 'medium',
};
