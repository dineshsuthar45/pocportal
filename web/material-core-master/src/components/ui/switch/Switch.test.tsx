import { render } from "@testing-library/react";
import React from "react";
import { MSwitch } from "./Switch";
import { SwitchProps } from "./Switch.types";

describe("Switch component", () => {
  const renderComponent = () => render(<MSwitch {...props} />);
  let props: SwitchProps;

  beforeEach(() => {
    props = {
      label: "Place Order",
      id: "sldr1",
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
