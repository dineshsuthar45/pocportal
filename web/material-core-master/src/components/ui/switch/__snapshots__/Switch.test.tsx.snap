// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component should render correctly 1`] = `
<div>
  <div
    class="switch-root light"
  >
    <div
      class="m-form-control-label end"
    >
      <div
        aria-label=""
        class="m-tooltip-root tooltip-label tooltip-container"
        data-mui-internal-clone-element="true"
      >
        <label>
          Place Order
        </label>
      </div>
      <div
        class="label-children"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary css-priqlz-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              id="sldr1"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
